dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/udp/UDPConfigurationImpl.java

```diff
@@ -0,0 +1,467 @@
+/*******************************************************************************
+ * Copyright (c) 2003, 2006 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.netty.internal.udp;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.ibm.websphere.channelfw.ChannelData;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelOption;
+import io.openliberty.netty.internal.BootstrapConfiguration;
+import io.openliberty.netty.internal.exception.NettyException;
+
+/**
+ * TODO: as far as I can tell UDP options aren't actually exposed to user config in Liberty, so currently a few props 
+ * (notably access lists) aren't implemented.
+ * 
+ * @see com.ibm.ws.udpchannel.internal.UDPChannelConfiguration
+ */
+public class UDPConfigurationImpl implements BootstrapConfiguration {
+
+    private static final TraceComponent tc = Tr.register(UDPConfigurationImpl.class, UDPMessageConstants.NETTY_TRACE_NAME, UDPMessageConstants.UDP_BUNDLE);
+
+    private static int DEFAULT_READ_BUFFER_SIZE = 1024000;
+    private Map<String, Object> channelProperties = null;
+
+
+    private ChannelData channelData;
+    private String hostname = null;
+    private int port = 0;
+    private int sendBufferSize = 0;
+    private int receiveBufferSize = 0;
+    private int channelReceiveBufferSize = UDPConfigConstants.MAX_UDP_PACKET_SIZE;
```

Comment:
As we discussed - this is confusing because the UDP channel keeps track of both a `receiveBufferSize` and a `channelReceiveBufferSize`. `receiveBufferSize` maps to `RECEIVE_BUFFER_SIZE_MAX`, and `channelReceiveBufferSize` maps to `MAX_UDP_PACKET_SIZE`.

Reply:
Earlier I proposed changing this `MAX_UDP_PACKET_SIZE` to something like `CHANNEL_RECEIVE_BUFFER_SIZE_MAX`, but looking at this again I think there's some value in leaving it as `MAX_UDP_PACKET_SIZE`